src/ray/util/util.h

```diff
@@ -155,6 +157,32 @@ inline std::string GenerateUUIDV4() {
   return ss.str();
 }
 
+/// Compress the input data using zstd compression
+inline std::string CompressZstd(const std::string &data) {
+  // Compress the data.
+  const size_t compressed_size = ZSTD_compressBound(data.size());
+  std::string compressed_data(compressed_size, '\0');
+  const size_t actual_compressed_size =
+      ZSTD_compress(compressed_data.data(), compressed_size, data.data(), data.size(), 1);
+  compressed_data.resize(actual_compressed_size);
+  return compressed_data;
+}
+
+/// Decompress the input data compressed by zstd compression
+inline std::string DecompressZstd(const std::string &data) {
+  // Get the decompressed size.
+  const size_t decompressed_size = ZSTD_getFrameContentSize(data.data(), data.size());
+  // Decompress the data.
+  std::string decompressed_data(decompressed_size, '\0');
```

Comment:
qq. Could we just initialized with size without copying the second parameter to the allocated string?

Reply:
not for [std::string](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) that i'm aware. there's a `reserve` introduced since C++20, but here in ray it's in C++17 i think.

we could use new/unique_ptr but then it has to be copied into a string before return.